bundles/model/org.eclipse.smarthome.model.persistence.tests/src/test/java/org/eclipse/smarthome/model/persistence/extensions/PersistenceExtensionsTest.java

```diff
@@ -124,8 +126,13 @@ public void testMaximumSince() {
     @Test
     public void testAverageSince() {
         item.setState(new DecimalType(3025));
-        DecimalType average = PersistenceExtensions.averageSince(item, new DateMidnight(2003, 1, 1), "test");
-        assertEquals("2100", average.toString());
+        DateMidnight startStored = new DateMidnight(2003, 1, 1);
+        DateMidnight endStored = new DateMidnight(2012, 1, 1);
+        long storedInterval = endStored.getMillis() - startStored.getMillis();
+        long recentInterval = DateTime.now().getMillis() - endStored.getMillis();
+        double expected = (2007.4994 * storedInterval + 2518.5 * recentInterval) / (storedInterval + recentInterval);
+        DecimalType average = PersistenceExtensions.averageSince(item, startStored, "test");
+        assertEquals(expected, average.doubleValue(), 0.001);
```

Comment:
Is this test stable enough? If the execution is slow, the average might be calculated with a much bigger value than recentInterval already and the results might differ, right?